Content.Server/Atmos/EntitySystems/FlammableSystem.cs

```diff
@@ -433,20 +433,40 @@ public override void Update(float frameTime)
 
                 if (flammable.FireStacks > 0)
                 {
-                    var air = _atmosphereSystem.GetContainingMixture(uid);
+                    var air = _atmosphereSystem.GetContainingMixture(uid, false, true);
 
-                    // If we're in an oxygenless environment, put the fire out.
-                    if (air == null || air.GetMoles(Gas.Oxygen) < 1f)
+                    if (flammable.RequireOxygen)
                     {
-                        Extinguish(uid, flammable);
-                        continue;
+                        // If we're in an oxygenless environment, put the fire out.
+                        if (air == null || air.GetMoles(Gas.Oxygen) < 1f)
+                        {
+                            Extinguish(uid, flammable);
+                            continue;
+                        }
                     }
 
                     var source = EnsureComp<IgnitionSourceComponent>(uid);
                     _ignitionSourceSystem.SetIgnited((uid, source));
 
                     if (TryComp(uid, out TemperatureComponent? temp))
-                        _temperatureSystem.ChangeHeat(uid, 12500 * flammable.FireStacks, false, temp);
+                    {
+                        // get hot
+                        var maxDeltaT = flammable.PeakFlameTemperature - temp.CurrentTemperature;
+                        if (maxDeltaT > 0) // we dont want to burn and get colder.
+                        {
+                            var tempBump = MathF.Sqrt(maxDeltaT) * flammable.FireStacks * 25; //monotonic function, slows as it reaches maxT
+                            _temperatureSystem.ChangeHeat(uid, tempBump, false, temp);
```

Comment:
`tempBump` is a measure of temperature, not heat?

Reply:
Ah am I missing a conversion from joules to kelvin somewhere? That could be.

Reply:
I think I addressed the rest of the reviews, but Im still thinking about this one.

notes to myself:
- Need to see if theres a k->j conversion function already
- check if it matters since I think I effectively abstracted it via the sqrt(diff)

Reply:
You just need Joules, see above.

Reply:
Done!